sources/api/thar-be-updates/src/status.rs

```diff
@@ -0,0 +1,352 @@
+use crate::error;
+use crate::error::Result;
+use bottlerocket_release::BottlerocketRelease;
+use chrono::{DateTime, Utc};
+use model::modeled_types::FriendlyVersion;
+use serde::{Deserialize, Serialize};
+use signpost::State;
+use snafu::{ensure, OptionExt, ResultExt};
+use std::convert::TryInto;
+use std::os::unix::process::ExitStatusExt;
+use std::process::Output;
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub enum UpdateState {
+    Idle,
+    Available,
+    Staged,
+    Ready,
+}
+
+/// UpdateImage represents a Bottlerocket update image
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct UpdateImage {
+    arch: String,
+    version: semver::Version,
+    variant: String,
+}
+
+impl UpdateImage {
+    pub fn get_version(&self) -> &semver::Version {
```

Comment:
From what I've seen of Rust style:
```suggestion
    pub fn version(&self) -> &semver::Version {
```

same with other get_* functions.